server/hotrod/pom.xml

```diff
@@ -199,6 +208,50 @@
                </instructions>
             </configuration>
          </plugin>
+
+         <plugin>
+            <groupId>org.infinispan</groupId>
+            <artifactId>infinispan-protocol-parser-generator-maven-plugin</artifactId>
+            <version>${project.version}</version>
+            <configuration>
+               <files>
+                  <file>${basedir}/src/main/resources/hotrod.gr</file>
+               </files>
+            </configuration>
+            <executions>
+               <execution>
+                  <goals>
+                     <goal>generate</goal>
+                  </goals>
+               </execution>
+            </executions>
+         </plugin>
+         <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>build-helper-maven-plugin</artifactId>
+            <executions>
+               <execution>
+                  <id>add-source</id>
+                  <phase>generate-sources</phase>
+                  <goals>
+                     <goal>add-source</goal>
+                  </goals>
+                  <configuration>
+                     <sources>
+                        <source>${project.build.directory}/generated-sources/java/</source>
```

Comment:
I think you can do the same thing build-helper-maven-plugin does in your plugin:

https://github.com/mojohaus/build-helper-maven-plugin/blob/master/src/main/java/org/codehaus/mojo/buildhelper/AddSourceMojo.java#L64